ktor-http-cio/src/io/ktor/http/cio/Multipart.kt

```diff
@@ -148,10 +147,10 @@ fun parseMultipart(
     boundaryBytes.put(pch)
     boundaryBytes.put(pch)
 
-    for (i in 0 until boundaryLength) {
-        val ch = contentType[boundaryStart + i].toInt() and 0xffff
-        if (ch > 0x7f) throw IOException("Failed to parse multipart: wrong boundary byte 0x${ch.toString(16)} - should be 7bit character")
-        boundaryBytes.put(ch.toByte())
+    for (ch in boundaryParameter) {
+        val b =  ch.toByte()
+        if (b > 0x7f) throw IOException("Failed to parse multipart: wrong boundary byte 0x${b.toString(16)} - should be 7bit character")
```

Comment:
Byte is signed `[-128, 127]` -- `0x7f == 127`. So I guess it doesn't work as expected. Maybe a test verifying that this exception is thrown if not there already?

`if (b > 0x7f)` -> `if ((b.toInt() and 0xFF) > 0x7f)`
or
`if (b > 0x7f)` -> `if (b < 0)`